--- src/Composer/Compiler.php.orig	2013-03-28 14:17:00.861834974 +0100
+++ src/Composer/Compiler.php	2013-03-28 14:17:20.158095673 +0100
@@ -81,9 +81,9 @@
             ->ignoreVCS(true)
             ->name('*.php')
             ->exclude('Tests')
-            ->in(__DIR__.'/../../vendor/symfony/')
-            ->in(__DIR__.'/../../vendor/seld/jsonlint/src/')
-            ->in(__DIR__.'/../../vendor/justinrainbow/json-schema/src/')
+            ->in('/usr/share/pear/Symfony/Component/')
+            ->in('/usr/share/php/Seld/')
+            ->in('/usr/share/php/JsonSchema/')
         ;
 
         foreach ($finder as $file) {
