diff --git src/Composer/Compiler.php src/Composer/Compiler.php
index 58c4d85..a794c36 100644
--- src/Composer/Compiler.php
+++ src/Composer/Compiler.php
@@ -189,12 +189,6 @@ Phar::mapPhar('composer.phar');
 
 EOF;
 
-        // add warning once the phar is older than 30 days
-        if (preg_match('{^[a-f0-9]+$}', $this->version)) {
-            $warningTime = time() + 30*86400;
-            $stub .= "define('COMPOSER_DEV_WARNING_TIME', $warningTime);\n";
-        }
-
         return $stub . <<<'EOF'
 require 'phar://composer.phar/bin/composer';
 
diff --git src/Composer/Console/Application.php src/Composer/Console/Application.php
index 22fc763..0380505 100755
--- src/Composer/Console/Application.php
+++ src/Composer/Console/Application.php
@@ -94,12 +94,6 @@ class Application extends BaseApplication
             $output->writeln('<warning>Composer only officially supports PHP 5.3.2 and above, you will most likely encounter problems with your PHP '.PHP_VERSION.', upgrading is strongly recommended.</warning>');
         }
 
-        if (defined('COMPOSER_DEV_WARNING_TIME') && $this->getCommandName($input) !== 'self-update') {
-            if (time() > COMPOSER_DEV_WARNING_TIME) {
-                $output->writeln(sprintf('<warning>Warning: This development build of composer is over 30 days old. It is recommended to update it by running "%s self-update" to get the latest version.</warning>', $_SERVER['PHP_SELF']));
-            }
-        }
-
         if (getenv('COMPOSER_NO_INTERACTION')) {
             $input->setInteractive(false);
         }
