diff --git core/composer.prov core/composer.prov
index c3f89be..4524e72 100755
--- composer.prov
+++ composer.prov
@@ -27,6 +27,7 @@ Taken from composer.json.  See http://getcomposer.org/doc/04-schema.md for detai
 
 import json
 import os
+import subprocess
 import sys
 
 paths = [path.rstrip() for path in sys.stdin.readlines()]
@@ -58,3 +59,7 @@ for path in paths:
         print '=', version
     else:
         print
+
+# Invoke the regular RPM provides generator to allow compatibility with RPM < 4.9 (no fileattrs)
+p = subprocess.Popen(['/usr/lib/rpm/find-provides'], stdout=subprocess.PIPE, stdin=subprocess.PIPE)
+print p.communicate(input='\n'.join(paths))[0]
diff --git core/macros.composer core/macros.composer
index d079cce..4103ae7 100644
--- macros.composer
+++ macros.composer
@@ -14,5 +14,9 @@
 
 %composer_install  %{_rpmconfigdir}/composer-install --composer="%__composer" --composer_dir="%composer" --php_dir="%{_datadir}/php" --bin_dir="%{_bindir}"
 
-# No-op macro to allow spec compatibility with RPM < 4.9 (no fileattrs)
-%composer_find_provides_and_requires %{nil}
+# Macro to allow spec compatibility with RPM < 4.9 (no fileattrs)
+%composer_find_provides_and_requires %%{expand: \
+%global _use_internal_dependency_generator 0
+%global __find_provides %{_rpmconfigdir}/composer.prov
+%global __find_requires %{_rpmconfigdir}/composer.req
+}
