diff --git a/src/Composer/Compiler.php b/src/Composer/Compiler.php
index 58c4d85..b6260bb 100644
--- src/Composer/Compiler.php
+++ src/Composer/Compiler.php
@@ -35,16 +35,7 @@ class Compiler
             unlink($pharFile);
         }
 
-        $process = new Process('git log --pretty="%H" -n1 HEAD', __DIR__);
-        if ($process->run() != 0) {
-            throw new \RuntimeException('Can\'t run git log. You must ensure to run compile from composer git repository clone and that git binary is available.');
-        }
-        $this->version = trim($process->getOutput());
-
-        $process = new Process('git describe --tags HEAD');
-        if ($process->run() == 0) {
-            $this->version = trim($process->getOutput());
-        }
+        $this->version = '__PHAR_VERSION__';
 
         $phar = new \Phar($pharFile, 0, 'composer.phar');
         $phar->setSignatureAlgorithm(\Phar::SHA1);
